Krishnan0022/Tic-Tac-Toe-Python
# Create README.md

Pull Request Title: Added AI and Two-Player Game Modes

Description:

This pull request introduces the following updates and features to the Tic Tac Toe Python game:

Single Player Mode (AI):

Implemented an AI opponent using the Minimax algorithm. The AI plays optimally and challenges the player during the game.
Players can now choose their symbol (X or O) and play against the AI.
Two Player Mode:

Added support for a local two-player mode, where two players can take turns on the same machine.
Players can choose their own symbols at the beginning of the game.
Game Flow Enhancements:

Improved the board display to make it more user-friendly and visually clear after each move.
Players are prompted to input their moves using a clear format (e.g., A1, B2, etc.).
The game announces the winner or a draw after the game concludes.
Minor Bug Fixes:

Fixed input validation for invalid moves.
Enhanced the user interface for better interaction and responsiveness.
Changes Made:

Updated the game logic to allow for both single-player and two-player modes.
Introduced the minimax() function to calculate optimal moves for the AI.
Refined board display and user input methods for better usability.
How to Test:

Run the game in the terminal by executing python ttt.py.
Choose between Single Player or Two Player mode.
Select your symbol (X or O) and follow the prompts to play the game.
Check that the game ends with a winner or draw and that the board displays correctly after each move.
Additional Notes:

Please make sure to review the AI logic for any potential edge cases.
The game has been designed to run on Python 3.x
